tests/yoti-common.anchor.processor.spec.js

```diff
@@ -0,0 +1,79 @@
+const fs = require('fs');
+const AnchorProcessor = require('../src/yoti_common/anchor.processor').AnchorProcessor;
+const protoRoot = require('../src/proto-root').initializeProtoBufObjects();
+const forge = require('node-forge');
+const ByteBuffer = require('bytebuffer');
+const expect = require('chai').expect;
+
+function parseAnchorData(anchorString) {
+  const decodedData = protoRoot.decodeEncryptedData(anchorString);
+  const cipherText = forge.util.decode64(decodedData.cipherText);
+  const AnchorBuf = Buffer.from(cipherText, 'binary');
+  const byteBuffer = ByteBuffer.wrap(AnchorBuf);
+  const AnchorObj = {
+    originServerCerts: [byteBuffer],
+    getSignedTimeStamp() {},
+  };
+
+  return AnchorObj;
+}
+
+describe('anchorProcessor', () => {
+  describe('#process', () => {
+    context('when processing DL Source Anchor data', () => {
+      const dlSourceAnchor = fs.readFileSync('./tests/sample-data/yoti-common/dl-source-anchor.txt', 'utf8');
+
+      it('should return DRIVING_LICENCE as value', () => {
+        const expectedSourceValue = 'DRIVING_LICENCE';
+        const dlAnchorObj = parseAnchorData(dlSourceAnchor);
+        const anchorSources = AnchorProcessor.process([dlAnchorObj]);
+        const anchorValue = anchorSources.sources[0].getValue();
+
+        expect(anchorValue).to.equal(expectedSourceValue);
+      });
+    });
+
+    context('when processing Passport Source Anchor data', () => {
+      const ppSourceAnchor = fs.readFileSync('./tests/sample-data/yoti-common/pp-source-anchor.txt', 'utf8');
+
+      it('should return PASSPORT as value', () => {
+        const expectedSourceValue = 'PASSPORT';
+        const ppAnchorObj = parseAnchorData(ppSourceAnchor);
+        const anchorSources = AnchorProcessor.process([ppAnchorObj]);
+        const anchorValue = anchorSources.sources[0].getValue();
+
+        expect(anchorValue).to.equal(expectedSourceValue);
+      });
+    });
+
+    context('when processing duplicate Source Anchors', () => {
```

Comment:
this is not expected behaviour, so you might want to get rid of this test and update the logic to allow duplicate values, as you can have duplicate values with different sub types (e.g. PASSPORT + NFC and PASSPORT + OCR)